## Programa Ingenias+ Data Science

## Trabajando con Pandas

1) Importa las librerias requeridas: `Pandas` y `NumPy`.  

Una buena practica a la hora de escribir codigo es de comentar siempre que estoy haciendo (usando `#`) en la linea anterior. Por mas obvio que parezca lo que voy a hacer, puede ser que tenga que compartir el documento con otras personas y no sepan que quise hacer en cada linea. O que yo mismo lea el codigo tiempo despues y no recuerde porque lo hice.

In [101]:
# Importamos Pandas
import pandas as pd

# Importamos Numpy
import numpy as np

2) Lee el archivo csv `life_expectancy_data` usando `pandas`. Guardalo en una variable llamada `expvida`.

In [102]:
# Leo los datos en un DataFrame
expvida = pd.read_csv('life_expectancy_data.csv')
expvida

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

In [103]:
# Leo los datos en un DataFrame
expvida = pd.read_csv('life_expectancy_data.csv')
expvida

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

3) ¿Que tipo de estructura de datos contiene la variable `expvida`?

In [104]:
# Obtengo el tipo de estructura de datos
type(expvida)

pandas.core.frame.DataFrame

4) ¿Cuantas filas y columnas tiene `expvida`?

In [105]:
# Uso .shape para obtener cantidad de filas y columnas
expvida.shape

(2938, 22)

**Numero de filas**: 2938

**Numero de columnas**: 22

5) ¿Cual es el nombre de las columnas contenidas en `expvida`?

In [106]:
# Obtengo el nombre de las columnas
expvida.columns

Index(['Country', 'Year', 'Status', 'Life expectancy ', 'Adult Mortality',
       'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B',
       'Measles ', ' BMI ', 'under-five deaths ', 'Polio', 'Total expenditure',
       'Diphtheria ', ' HIV/AIDS', 'GDP', 'Population',
       ' thinness  1-19 years', ' thinness 5-9 years',
       'Income composition of resources', 'Schooling'],
      dtype='object')

6) Inspecciona las primeras 10 filas de `expvida`

In [107]:
# Inspecciono las primeras filas
expvida[:10]

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0            263.0   
1  Afghanistan  2014  Developing              59.9            271.0   
2  Afghanistan  2013  Developing              59.9            268.0   
3  Afghanistan  2012  Developing              59.5            272.0   
4  Afghanistan  2011  Developing              59.2            275.0   
5  Afghanistan  2010  Developing              58.8            279.0   
6  Afghanistan  2009  Developing              58.6            281.0   
7  Afghanistan  2008  Developing              58.1            287.0   
8  Afghanistan  2007  Developing              57.5            295.0   
9  Afghanistan  2006  Developing              57.3            295.0   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624         65.0      1154  ...   
1             64     0.01               73.523582         62.0       492  ...   
2             66     0.01               73.219243         64.0       430  ...   
3             69     0.01               78.184215         67.0      2787  ...   
4             71     0.01                7.097109         68.0      3013  ...   
5             74     0.01               79.679367         66.0      1989  ...   
6             77     0.01               56.762217         63.0      2861  ...   
7             80     0.03               25.873925         64.0      1599  ...   
8             82     0.02               10.910156         63.0      1141  ...   
9             84     0.03               17.171518         64.0      1990  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0    6.0               8.16         65.0        0.1  584.259210  33736494.0   
1   58.0               8.18         62.0        0.1  612.696514    327582.0   
2   62.0               8.13         64.0        0.1  631.744976  31731688.0   
3   67.0               8.52         67.0        0.1  669.959000   3696958.0   
4   68.0               7.87         68.0        0.1   63.537231   2978599.0   
5   66.0               9.20         66.0        0.1  553.328940   2883167.0   
6   63.0               9.42         63.0        0.1  445.893298    284331.0   
7   64.0               8.33         64.0        0.1  373.361116   2729431.0   
8   63.0               6.73         63.0        0.1  369.835796  26616792.0   
9   58.0               7.43         58.0        0.1  272.563770   2589345.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   
5                   18.4                 18.4   
6                   18.6                 18.7   
7                   18.8                 18.9   
8                   19.0                 19.1   
9                   19.2                 19.3   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  
5                            0.448        9.2  
6                            0.434        8.9  
7                            0.433        8.7  
8                            0.415        8.4  
9                            0.405        8.1  

[10 rows x 22 columns]

7) Y ahora las 10 ultimas.

In [108]:
# Inspecciono las ultimas filas
expvida.tail(10)

Country  Year      Status  Life expectancy   Adult Mortality  \
2928  Zimbabwe  2009  Developing              50.0            587.0   
2929  Zimbabwe  2008  Developing              48.2            632.0   
2930  Zimbabwe  2007  Developing              46.6             67.0   
2931  Zimbabwe  2006  Developing              45.4              7.0   
2932  Zimbabwe  2005  Developing              44.6            717.0   
2933  Zimbabwe  2004  Developing              44.3            723.0   
2934  Zimbabwe  2003  Developing              44.5            715.0   
2935  Zimbabwe  2002  Developing              44.8             73.0   
2936  Zimbabwe  2001  Developing              45.3            686.0   
2937  Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
2928             30     4.64                1.040021         73.0       853   
2929             30     3.56               20.843429         75.0         0   
2930             29     3.88               29.814566         72.0       242   
2931             28     4.57               34.262169         68.0       212   
2932             28     4.14                8.717409         65.0       420   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
2928  ...   69.0               6.26         73.0       18.1   65.824121   
2929  ...   75.0               4.96         75.0       20.5  325.678573   
2930  ...   73.0               4.47         73.0       23.7  396.998217   
2931  ...   71.0               5.12          7.0       26.8  414.796232   
2932  ...   69.0               6.44         68.0       30.3  444.765750   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
2928   1381599.0                    7.5                  7.4   
2929  13558469.0                    7.8                  7.8   
2930   1332999.0                    8.2                  8.2   
2931  13124267.0                    8.6                  8.6   
2932    129432.0                    9.0                  9.0   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
2928                            0.419        9.9  
2929                            0.421        9.7  
2930                            0.414        9.6  
2931                            0.408        9.5  
2932                            0.406        9.3  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427       10.0  
2936                            0.427        9.8  
2937                            0.434        9.8  

[10 rows x 22 columns]

8) ¿Que tipos de datos contiene cada una de las columnas de `expvida`?

In [109]:
# Tipo de datos en cada columna
expvida.dtypes

Country                             object
Year                                 int64
Status                              object
Life expectancy                    float64
Adult Mortality                    float64
infant deaths                        int64
Alcohol                            float64
percentage expenditure             float64
Hepatitis B                        float64
Measles                              int64
 BMI                               float64
under-five deaths                    int64
Polio                              float64
Total expenditure                  float64
Diphtheria                         float64
 HIV/AIDS                          float64
GDP                                float64
Population                         float64
 thinness  1-19 years              float64
 thinness 5-9 years                float64
Income composition of resources    float64
Schooling                          float64
dtype: object

9) ¿Hay valores faltantes en alguna de las columnas de `expvida`?

In [110]:
# Obtengo la cantidad de datos faltantes en cada columna
expvida.isnull().sum()

Country                              0
Year                                 0
Status                               0
Life expectancy                     10
Adult Mortality                     10
infant deaths                        0
Alcohol                            194
percentage expenditure               0
Hepatitis B                        553
Measles                              0
 BMI                                34
under-five deaths                    0
Polio                               19
Total expenditure                  226
Diphtheria                          19
 HIV/AIDS                            0
GDP                                448
Population                         652
 thinness  1-19 years               34
 thinness 5-9 years                 34
Income composition of resources    167
Schooling                          163
dtype: int64

10) Selecciona el maximo de datos faltantes. Calcula que porcentaje de los datos representan los datos faltantes.

In [111]:
# Obtengo el maximo valor de datos faltantes y lo guardo en la variable masnan
datos_faltantes = expvida.isnull().sum()
masnan = datos_faltantes.max()

print("Máximo valor de datos faltantes ->", masnan)

Máximo valor de datos faltantes -> 652


In [112]:
# Obtengo la cantidad de filas total y divido el maximo valor de datos faltantes por este numero
# y lo multiplico por 100
porcentaje_faltantes = (masnan / len(expvida)) * 100

print("Porcentaje de datos faltantes -> ", porcentaje_faltantes)

Porcentaje de datos faltantes ->  22.19196732471069


In [113]:
# Imprimo el maximo porcentaje de datos faltantes. Uso f-strings.
print(f"Porcentaje de datos faltantes -> {porcentaje_faltantes:.2f}%")

Porcentaje de datos faltantes -> 22.19%


- str.format()
- Template Class
- f-strings

#### Ahora vamos a realizar algunos calculos y manipulaciones sobre las columnas.

Una función útil para este fin es: `.rename()`. Y podemos utilizarla de la siguiente manera: Si `df` es mi DataFrame y quiero renombrar la columna `columna` como `columna1`, y la columna `columna2` como `columna3`, utilizo la siguiente sintaxis:
  
`df.rename(columns={"columna": "columna1", "columna2": "columna3"}, inplace=True)`

11) Renombra la columna 'Life expectancy ' como 'life_expectancy', ' BMI ' como 'bmi' y ''Measles ' como 'measles'. No olvides de usar `inplace` para modificar el DataFrame original. Luego chequea que el cambio se haya hecho correctamente.

In [114]:
# Renombro columnas del DataFrame
expvida.rename(columns={'Life expectancy ':'life_expectancy', ' BMI ': 'bmi', 'Measles ':'measles'}, inplace = True)

In [115]:
# Chequeo que se hayan reemplazado correctamente
print(expvida.columns)

Index(['Country', 'Year', 'Status', 'life_expectancy', 'Adult Mortality',
       'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B',
       'measles', 'bmi', 'under-five deaths ', 'Polio', 'Total expenditure',
       'Diphtheria ', ' HIV/AIDS', 'GDP', 'Population',
       ' thinness  1-19 years', ' thinness 5-9 years',
       'Income composition of resources', 'Schooling'],
      dtype='object')


12) ¿Cual es el promedio de expectativa de vida (Life expectancy) en general?

In [116]:
# Obtengo el promedio de la columna
promedio_vida = expvida['life_expectancy'].mean()

print("Promedio de expectativa de vida -> ", promedio_vida)

Promedio de expectativa de vida ->  69.22493169398908


13) ¿Cual es el promedio de expectativa de vida (Life expectancy) por país?

In [117]:
# Obtengo el promedio por pais
promedio_por_pais = expvida.groupby('Country')['life_expectancy'].mean()

promedio_por_pais

Country
Afghanistan                           58.19375
Albania                               75.15625
Algeria                               73.61875
Angola                                49.01875
Antigua and Barbuda                   75.05625
                                        ...   
Venezuela (Bolivarian Republic of)    73.38750
Viet Nam                              74.77500
Yemen                                 63.86250
Zambia                                53.90625
Zimbabwe                              50.48750
Name: life_expectancy, Length: 193, dtype: float64

14) ¿Cual es la expectativa de vida mas alta y mas baja? Usa la funcion de pandas y la funcion de NumPy y compara el resultado. ¿Que paises tienen esas expectativas de vida?

In [118]:
# Maxima expectativa de vida usando numpy
maxima_expectativa_numpy = np.max(expvida['life_expectancy'])

In [119]:
# Maxima expectativa de vida usando pandas
maxima_expectativa_pandas = expvida['life_expectancy'].max()

In [120]:
# Minima expectativa de vida usando numpy
minima_expectativa_numpy = np.min(expvida['life_expectancy'])

In [121]:
# Minima expectativa de vida usando pandas
minima_expectativa_pandas = expvida['life_expectancy'].min()

In [122]:
# Muestro los resultados maximos
print("Máxima expectativa de vida numpy -> ", maxima_expectativa_numpy)
print("Máxima expectativa de vida pandas -> ", maxima_expectativa_pandas)

Máxima expectativa de vida numpy ->  89.0
Máxima expectativa de vida pandas ->  89.0


In [123]:
# Muestro los resultados minimos
print("Mínima expectativa de vida numpy -> ", minima_expectativa_numpy)
print("Mínima expectativa de vida pandas -> ", minima_expectativa_pandas)

Mínima expectativa de vida numpy ->  36.3
Mínima expectativa de vida pandas ->  36.3


In [124]:
# Filtro las columnas para mostrar solo las filas que tengan valor
# igual al maximo
promedio_por_pais = expvida.groupby('Country')['life_expectancy'].mean()
valores_maximo = promedio_por_pais == promedio_por_pais.max()
paises_maximo = promedio_por_pais[valores_maximo]

print(paises_maximo)

Country
Japan    82.5375
Name: life_expectancy, dtype: float64


In [125]:
# Filtro las columnas para mostrar solo las filas que tengan valor
# igual al minimo
promedio_por_pais = expvida.groupby('Country')['life_expectancy'].mean()
valores_minimo = promedio_por_pais == promedio_por_pais.min()
paises_minimo = promedio_por_pais[valores_minimo]

print(paises_minimo)

Country
Sierra Leone    46.1125
Name: life_expectancy, dtype: float64


15) ¿Cuantas categorias encontramos en la columna "Status"? _**Pista**_: Lee acerca de la función [unique()](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.unique.html?highlight=unique#pandas.Series.unique)

In [126]:
# Obtengo los valores unicos de la columna status
valores_unicos = expvida['Status'].unique()

print("Valores únicos en la columna Status:")
print(valores_unicos)


Valores únicos en la columna Status:
['Developing' 'Developed']


16) ¿Cuantos paises se ubican en cada una de esas categorias de "Status"? _**Pista**_: Lee acerca de la función [value_counts()](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.value_counts.html)

In [127]:
# Cuento cuantos valores hay de cada categoria
paises_por_categoria = expvida['Status'].value_counts()

print(paises_por_categoria)

Status
Developing    2426
Developed      512
Name: count, dtype: int64


18) ¿Cuantos paises presentaron algun caso de Sarampion (**Measles**) en el 2015? **_Pista_**: Lee acerca de la función [.count()](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.count.html?highlight=count#pandas.DataFrame.count)

In [128]:
# Cuento cuantas filas tienen valores distintos de cero
expvida_2015 = expvida.query('Year == 2015')
expvida_2015

Country  Year      Status  life_expectancy  \
0                            Afghanistan  2015  Developing             65.0   
16                               Albania  2015  Developing             77.8   
32                               Algeria  2015  Developing             75.6   
48                                Angola  2015  Developing             52.4   
64                   Antigua and Barbuda  2015  Developing             76.4   
...                                  ...   ...         ...              ...   
2858  Venezuela (Bolivarian Republic of)  2015  Developing             74.1   
2874                            Viet Nam  2015  Developing             76.0   
2890                               Yemen  2015  Developing             65.7   
2906                              Zambia  2015  Developing             61.8   
2922                            Zimbabwe  2015  Developing             67.0   

      Adult Mortality  infant deaths  Alcohol  percentage expenditure  \
0               263.0             62     0.01               71.279624   
16               74.0              0     4.60              364.975229   
32               19.0             21      NaN                0.000000   
48              335.0             66      NaN                0.000000   
64               13.0              0      NaN                0.000000   
...               ...            ...      ...                     ...   
2858            157.0              9      NaN                0.000000   
2874            127.0             28      NaN                0.000000   
2890            224.0             37      NaN                0.000000   
2906             33.0             27      NaN                0.000000   
2922            336.0             22      NaN                0.000000   

      Hepatitis B  measles  ...  Polio  Total expenditure  Diphtheria   \
0            65.0     1154  ...    6.0               8.16         65.0   
16           99.0        0  ...   99.0               6.00         99.0   
32           95.0       63  ...   95.0                NaN         95.0   
48           64.0      118  ...    7.0                NaN         64.0   
64           99.0        0  ...   86.0                NaN         99.0   
...           ...      ...  ...    ...                ...          ...   
2858         87.0        0  ...   87.0                NaN         87.0   
2874         97.0      256  ...   97.0                NaN         97.0   
2890         69.0      468  ...   63.0                NaN         69.0   
2906          9.0        9  ...    9.0                NaN          9.0   
2922         87.0        0  ...   88.0                NaN         87.0   

       HIV/AIDS           GDP  Population   thinness  1-19 years  \
0           0.1    584.259210  33736494.0                   17.2   
16          0.1   3954.227830     28873.0                    1.2   
32          0.1   4132.762920  39871528.0                    6.0   
48          1.9   3695.793748   2785935.0                    8.3   
64          0.2  13566.954100         NaN                    3.3   
...         ...           ...         ...                    ...   
2858        0.1           NaN         NaN                    1.6   
2874        0.1           NaN         NaN                   14.2   
2890        0.1           NaN         NaN                   13.6   
2906        4.1   1313.889646    161587.0                    6.3   
2922        6.2    118.693830  15777451.0                    5.6   

       thinness 5-9 years  Income composition of resources  Schooling  
0                    17.3                            0.479       10.1  
16                    1.3                            0.762       14.2  
32                    5.8                            0.743       14.4  
48                    8.2                            0.531       11.4  
64                    3.3                            0.784       13.9  
...                   ...                              ...        ...